recipe/meta.yaml

```diff
@@ -13,30 +13,30 @@ source:
       - patches/0001-find_numpy.patch
 
 build:
-  number: 1
+  number: 2
   skip: True  # [win and vc<14]
 
 requirements:
   build:
     - python                                 # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - numpy                                  # [build_platform != target_platform]
-    - pybind11                               # [build_platform != target_platform]
-    - {{ compiler('cxx') }}
+    - pybind11 >=2.12.0,<3                   # [build_platform != target_platform]
+    - {{ compiler("cxx") }}
+    - {{ stdlib("c") }}
     - cmake
     - make                                   # [unix]
   host:
     - python
     - xtl >=0.7,<0.8
     - xtensor >=0.25.0,<0.26
-    - pybind11 >=2.6.1,<3
+    - pybind11 >=2.12.0,<3
```

Comment:
This is done to make sure a NumPy 2 compatible `pybind11` version is used (so `2.12.0+`)